tests/integration/test_random_advanced.py

```diff
@@ -173,6 +181,81 @@ def test_negative_binomial(t):
     assert_distribution(a, theo_mean, theo_std)
 
 
+FUNC_ARGS = (
+    ("binomial", (15, 0.666)),
+    ("negative_binomial", (15, 0.666)),
+    ("geometric", (0.707,)),
+    ("hypergeometric", (60, 440, 200)),
+    ("standard_t", (3.1415,)),
+    ("vonmises", (1.414, 3.1415)),
+    ("wald", (1.414, 3.1415)),
+    ("zipf", (7.5,)),
+)
+
+
+@pytest.mark.parametrize("func, args", FUNC_ARGS, ids=str)
+@pytest.mark.parametrize("size", ((2048 * 2048), (4096,), 25535), ids=str)
+def test_random_sizes(func, args, size):
+    seed = 42
+    gen_np = np.random.Generator(np.random.PCG64(seed=seed))
+    gen_num = num.random.Generator(num.random.XORWOW(seed=seed))
+    a_np = getattr(gen_np, func)(*args, size=size)
+    a_num = getattr(gen_num, func)(*args, size=size)
+    assert a_np.shape == a_num.shape
+
+
+@pytest.mark.xfail
+@pytest.mark.parametrize("func, args", FUNC_ARGS, ids=str)
+def test_random_size_none(func, args):
+    seed = 42
+    gen_np = np.random.Generator(np.random.PCG64(seed=seed))
+    gen_num = num.random.Generator(num.random.XORWOW(seed=seed))
+    a_np = getattr(gen_np, func)(*args, size=None)
+    a_num = getattr(gen_num, func)(*args, size=None)
+    # cuNumeric returns singleton array
+    # NumPy returns scalar
```

Comment:
Based on this comment:

https://github.com/nv-legate/cunumeric/pull/347#discussion_r870879150

Is this a case where this is an expected difference? If so, we should update the test to codify that difference (here and elsewhere) cc @manopapad

Reply:
I believe this more like what happened to random_integers which was fixed in e1ed04cbf8f7b6dc0d29a86813890295972d6d15

I intentionally avoided using `isscalar()` in these tests, but cuNumeric is just not returning scalars, "deferred" or not.

But I'll leave this to Manolis for a definitive answer.

Reply:
This goes beyond our known incompatibility when returning scalars (where we return 0d arrays in some cases where NumPy would return a python scalar). Using `np.ndim(.) == 0` instead of `np.isscalar(.)` should work in all cases.

This is pointing to a real bug in cuNumeric. The functions in `bitgenerator.py` are using the wrong default shape `(1,)` instead of `()`, e.g. see https://github.com/nv-legate/cunumeric/blob/branch-23.07/cunumeric/random/bitgenerator.py#L600. We should probably make this change uniformly across the file (and check that it indeed works).

Reply:
@yimoj can you make a GH issue to follow up on this?

Reply:
@bryevdv I have this tracked in https://github.com/nv-legate/cunumeric/issues/991, need me to split the size one out and open a "bug bug" instead?

Reply:
@yimoj No I suppose one issue is enough